### PR TITLE
Enable env and arg to be set on tests.

### PR DESCRIPTION
**Pull request by n-g**
Fixes NSL-349, NSL-354

### CODE REVIEW

File: internal/parsing/env.go [hunk 1]
```diff
@@ -2,24 +2,27 @@
...
 import (
 	"google.golang.org/protobuf/proto"
 	"namespacelabs.dev/foundation/internal/fnerrors"
 	"namespacelabs.dev/foundation/schema"
 )
 
-func SetEnv(merged []*schema.BinaryConfig_EnvEntry, entry *schema.BinaryConfig_EnvEntry) ([]*schema.BinaryConfig_EnvEntry, error) {
-	for _, existing := range merged {
-		if entry.Name == existing.Name {
+func MergeEnvs(base []*schema.BinaryConfig_EnvEntry, additional []*schema.BinaryConfig_EnvEntry) ([]*schema.BinaryConfig_EnvEntry, error) {
```
Comment (hugosantos) [inline internal/parsing/env.go:13]: Add a test too? :-)

File: internal/frontend/cuefrontendopaque/tests.go [hunk 2]
```diff
@@ -48,9 +51,20 @@ func parseTest(ctx context.Context, env *schema.Environment, pl parsing.EarlyPac
...
 
+	envVars, err := bits.Env.Parsed(pkg.Location.PackageName)
+	if err != nil {
+		return nil, err
+	}
+
 	out := &schema.Test{
 		Name:             name,
 		ServersUnderTest: bits.Servers,
+		Driver: &schema.Binary{
```
Comment (hugosantos) [inline internal/frontend/cuefrontendopaque/tests.go:62]: It feels confusing to have a Driver which is not an actual binary. I suggest just having a schema.BinaryConfig here, which then gets merged as you're doing below.

File: internal/support/env.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	"google.golang.org/protobuf/proto"
+	"namespacelabs.dev/foundation/internal/fnerrors"
+	"namespacelabs.dev/foundation/schema"
+)
+
+func MergeEnvs(base []*schema.BinaryConfig_EnvEntry, additional []*schema.BinaryConfig_EnvEntry) ([]*schema.BinaryConfig_EnvEntry, error) {
+	merged := append(base, additional...)
+
+	// Check for collisions.
+	cache := make(map[string]*schema.BinaryConfig_EnvEntry)
```
Comment (hugosantos) [inline internal/support/env.go:17]: This is not a cache, it's an index. :-)

File: internal/support/env.go [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+	for _, entry := range merged {
```
Comment (hugosantos) [inline internal/support/env.go:18]: Isn't this the same implementation as parsing/env.go?

File: internal/testing/fixture.go [hunk 5]
```diff
@@ -150,6 +150,7 @@ func PrepareTest(ctx context.Context, pl *parsing.PackageLoader, env cfg.Context
 		Command:            testBin.Command,
 		Args:               testBin.Args,
 		WorkingDir:         testBin.WorkingDir,
+		Env:                testDef.Driver.Config.Env,
```
Comment (hugosantos) [inline internal/testing/fixture.go:153]: This doesn't seem consistent; above we use testBin.Args, and here only Config.Env? Where's the merged ones?

File: internal/support/env.go (hunk 4 above)
Comment (n-g) [reply internal/support/env.go:18]: Was supposed to be a move.
Can't live here due to dep cycle. Cleaned up.

File: internal/testing/fixture.go (hunk 5 above)
Comment (n-g) [reply internal/testing/fixture.go:153]: Fixed with https://github.com/namespacelabs/foundation/commit/5e91f09ab775eae835dbf4eac5dbbc9e1529dcdc
and aligned other places that did not propagate env.

**Review by gmichelo — APPROVED** [review]

**Pull request merged**